src/d/llvm/expression.d

```diff
@@ -353,13 +353,19 @@ struct ExpressionGen {
 			
 			case Minus :
 				import d.llvm.type;
-				return LLVMBuildSub(
-					builder,
-					LLVMConstInt(TypeGen(pass.pass).visit(e.type), 0, true),
-					visit(e.expr),
-					"",
-				);
-			
+				import d.common.builtintype : isFloat;
+				if (e.expr.type.builtin.isFloat())
+				{
+					return LLVMBuildFNeg(builder, visit(e.expr), "");
```

Comment:
IMO, we'd be better off using different operator at the IR level rather than switching here. float and int are going to be treated differently at every stage of the process.